raiden-ts/src/raiden.ts

```diff
@@ -1474,20 +1474,30 @@ export class Raiden {
             mergeMap(async (response) => response.text()),
             map((text) => {
               const suggestions = decode(SuggestedPartners, jsonParse(text));
-              assert(suggestions.length, ['empty partner suggestion from PFS', { pfs: pfs.url }]);
+
+              if (!suggestions.length) {
+                firstResponse = [];
+                throw Error();
```

Comment:
There's a log call on the `catchError`, it may be interesting to put some message here to get logged on this case.

Reply:
`throw` must not be conditional to `!firstResponse`, or else it won't retry the next PFS if the array is empty but `firstResponse` is already set. The whole block should be something like:
```
if (!suggestions.length) {
  if (!firstResponse) firstResponse = suggestions;
  throw new Error('your message to be logged');
}
return suggestions;
```

Reply:
Or, if you wanna abuse the `throw` operator to reuse the log and var set inside the catchError block:
```ts
if (!suggestions.length) throw suggestions; // throw empty array to be set to firstResponse
return suggestions;
```

Reply:
We cannot use `(!firstResponse)` as a condition because that would mean an error from PFS1 would never be overwritten by an empty result of PFS2.

Reply:
This is fine, `firstResponse` is **first** response, either an error or an empty array.

Reply:
But I think then it's already nitpick, I think both behaviors are acceptable, so leave it as you find better.

Reply:
But our priority is to rather return an empty list than an error, because that means more information for the user. So we want to overwrite an earlier error with an empty list.

This is what https://github.com/raiden-network/light-client/pull/2491/files#diff-7c34034053ef8211fe51e9a7d5693c5b6249d7183ea9d8ce4d7327537e2b700aR1640 tests.